views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js

```diff
@@ -303,10 +377,60 @@ define([
     }
 
     MathEntryInteractionStateResponse.prototype.inGapMode = function inGapMode() {
-        var interaction = this.widget.element;
-        var useGapExpression = interaction.prop('useGapExpression');
+        const interaction = this.widget.element;
+        const useGapExpression = interaction.prop('useGapExpression');
         return useGapExpression && useGapExpression !== 'false' || false;
     }
 
+    MathEntryInteractionStateResponse.prototype.initAlternativeInput = function initAlternativeInput() {
+        $addAlternativeBtn.on('click', () => {
+            this.addAlternativeInput(0)
+        });
+    }
+
+    MathEntryInteractionStateResponse.prototype.addAlternativeInput = function addAlternativeInput(responseId = 0) { 
+        const interaction = this.widget.element;
+        const $container = this.widget.$container;
+        const response = interaction.getResponseDeclaration();
+        const mapEntries = response.getMapEntries();
+        let dataIndex = 0;
+        const focusSelected = $container.find('.math-entry-input');
+        if (focusSelected.length > 0) {
+            dataIndex = focusSelected[focusSelected.length - 1].parentNode.dataset.index;
+        }
+
+        let responseValue = '';
+
+        let mapEntryResponse = Object.keys(mapEntries);
+        let gapValues = mapEntryResponse[responseId] ? this.getGapResponseObject(mapEntryResponse[responseId]) : null;
+        if (this.inGapMode() === true) { 
+            responseValue = this.gapTemplate
+        } else {
+            responseValue = this.correctResponses[responseId]
+        }
+
+        $('button.math-entry-response-correct').before(alternativeFormTpl({
```

Comment:
**issue:** Selectors need to be scoped, otherwise they may collide with the upper scope.

**suggestion:** scope the selection with the container
```suggestion
        $('button.math-entry-response-correct', $container).before(alternativeFormTpl({
```